src/components/molecules/SecretPasswordForm/SecretPasswordForm.jsx

```diff
@@ -26,6 +26,7 @@ const SecretPasswordForm = ({ buttonText = "Join the party" }) => {
   const passwordSubmitted = useCallback(
     async (e) => {
       e.preventDefault();
+      setInvalidPassword(false);
```

Comment:
Why was this moved up here? In the old position, it would clear it when the check succeeded, which seems more logical than clearing it just because we submitted the check?

`passwordChanged` already seems to clear invalid passwords as well.

Reply:
If you enter an incorrect password, and then type a new one, the message won't disappear until the requests succeeds. I moved the code at the top to make the message disappear the moment you send the request. By default `invalidPassword` is `false` and this won't trigger a re-render.

Reply:
Weird, the passwordChanged function reads as though it will reset it. I assumed that would happen as the new password is typed, but I guess not.

Reply:
I'll double check just in case.

Reply:
Sounds good, thanks :)